cookbook/forms/focus.md

```diff
@@ -0,0 +1,187 @@
+---
+layout: page
+title: "Focus on a Text Field"
+permalink: /cookbook/forms/focus/
+---
+
+When a text field is selected and accepting input, it is said to have "focus." 
+Generally, users can focus text fields by tapping on them, and developers 
+can focus text fields using the tools described in this recipe. 
+
+Managing focus is fundamental tool for creating forms with an intuitive 
+flow. For example, say we have a search screen with a text field. When 
+the user navigates to the search screen, we can focus the search term text field. 
+This allows the user to start typing as soon as the screen 
+is visible, without needing to manually tap on the text field!
+
+In this recipe, we'll learn how to focus a text field as soon as it's visible
+as well as how to focus a text field when a button is tapped.
+
+## Focus a text field as soon as it's visible
+
+In order to focus a text field as soon as it's visible, we can use the 
+`autofocus` property.
+
+```dart
+new TextField(
+  autofocus: true,
+);
+```
+
+For more information on handling input and creating text fields, please see the
+[Forms section of the cookbook](/cookbook#forms).
+
+## Focus a text field when a button is tapped
+
+Rather than immediately focusing a specific text field, we might need to focus a 
+text field at a later point in time. In this example, we'll see how to focus a 
+text field after the user presses a button. In the real world, you may also need 
+to focus a specific text field in response to an api call or a validation error.
+
+### Directions
+
+  1. Create a `FocusNode`
+  2. Pass the `FocusNode` to a `TextField`
+  3. Focus the `TextField` when a button is tapped
+
+### 1. Create a `FocusNode`
+
+First, we'll need to create a [`FocusNode`](https://docs.flutter.io/flutter/widgets/FocusNode-class.html).
+We will use the `FocusNode` to identify a specific `TextField` in Flutter's 
+"focus tree." This will allow us to focus the `TextField` in the next steps.
+
+Since focus nodes are long-lived objects, we need to store them in a `State` 
+class. In addition, we need to `dispose` of them when they're no longer needed! 
+
+```dart
+// Define a Custom Form Widget
+class MyForm extends StatefulWidget {
+  @override
+  _MyFormState createState() => new _MyFormState();
+}
+
+// Define a corresponding State class. This class will hold the data related to
+// the form.
+class _MyFormState extends State<MyForm> {
+  // Create the focus node. We will pass it to the TextField below.
+  final FocusNode myFocusNode = new FocusNode();
```

Comment:
Hrm, I could be thinking of this in the wrong way, but where else would we store the instance? We need to create the instance once and hold onto it for the lifecycle of the Widget. This allows to `requestFocus` with this instance of the focusNode, and `dispose` of it when the State is disposed as well.

What were you thinking of as an alternative?

Reply:
No sure, but I tend to think of stateful widgets with only finals as a bit of a code smell. As I'm not familiar with FocusNode, this might be totally fine, but wanted to check.